pymc3/plots/energyplot.py

```diff
@@ -47,15 +48,13 @@ def energyplot(trace, kind='kde', figsize=None, ax=None, legend=True, lw=0, alph
         _, ax = plt.subplots(figsize=figsize)
 
     if kind == 'kde':
-        for series in series_dict:
-            density, l, u = fast_kde(series_dict[series])
-            x = np.linspace(l, u, len(density))
-            ax.plot(x, density, label=series, **kwargs)
-            ax.fill_between(x, density, alpha=alpha)
-            
+        for label, value in series:
+            kdeplot(value, label=label, alpha=alpha, shade=True, ax=ax,
```

Comment:
Thanks for the comment. Not sure if I am really understanding you. I think this is what we always do, isn't it?